### PR TITLE
mgmt, fix Retry-After in LRO is not respected

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
However it appears that only the Retry-After header of the 1st response is used by `PollerFlux`.
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/util/polling/PollerFlux.java#L221

### CODE REVIEW

File: sdk/core/azure-core-management/src/test/java/com/azure/core/management/implementation/polling/LROPollerTests.java [hunk 1]
```diff
@@ -324,6 +274,102 @@ public String getName() {
         }
     }
 
+    @Test
+    public void lroRetryAfter() {
+        ServerConfigure configure = new ServerConfigure();
+        Duration expectedPollingDuration = Duration.ofSeconds(3);
+        configure.pollingCountTillSuccess = 3;
+        configure.additionalHeaders = new HttpHeaders(new HttpHeader("Retry-After", "1"));  // 1 second
```
Comment (weidongxu-microsoft) [inline sdk/core/azure-core-management/src/test/java/com/azure/core/management/implementation/polling/LROPollerTests.java:282]: Set `Retry-After` header to different value of default poll interval.

File: sdk/core/azure-core-management/src/test/java/com/azure/core/management/implementation/polling/LROPollerTests.java [hunk 2]
```diff
@@ -324,6 +276,107 @@ public String getName() {
...
+            FooWithProvisioningState result = lroFlux
+                .doOnNext(response -> {
+                    System.out.println(String.format("[%s] status %s",
+                        OffsetDateTime.now().toString(), response.getStatus().toString()));
+                }).blockLast()
+                .getFinalResult().block();
+            Assertions.assertNotNull(result);
+
+            Duration pollingDuration = Duration.ofNanos(System.nanoTime() - nanoTime);
+            Assertions.assertTrue(pollingDuration.compareTo(expectedPollingDuration) > 0);
```
Comment (ChenTanyi) [inline sdk/core/azure-core-management/src/test/java/com/azure/core/management/implementation/polling/LROPollerTests.java:312]: If we just check polling duration larger than expected. I think we could make `Retry-After` larger than the original one.

Comment (weidongxu-microsoft) [reply sdk/core/azure-core-management/src/test/java/com/azure/core/management/implementation/polling/LROPollerTests.java:312]: Yes. In this test, Retry-After is 1sec, default interval is 100ms.

Comment (ChenTanyi) [reply sdk/core/azure-core-management/src/test/java/com/azure/core/management/implementation/polling/LROPollerTests.java:312]: Got it. I had thought it was 30s.

File: sdk/core/azure-core-management/src/main/java/com/azure/core/management/implementation/polling/PollOperation.java [hunk 3]
```diff
@@ -45,35 +46,14 @@
...
+                return pollResponseMonoFromPollingState(serializerAdapter, pollResultType, pollingState);
             } else {
                 // InProgress|NonTerminal-Status
-                Mono<PollResponse<PollResult<T>>> pollResponse = pollResponseMono(serializerAdapter,
-                    pollingState.getOperationStatus(),
-                    pollingState.getLastResponseBody(),
-                    pollResultType);
                 return doSinglePoll(pipeline, pollingState)
-                    .doOnNext(updatedState -> updatedState.store(context))
-                    .then(pollResponse);
```
Comment (weidongxu-microsoft) [inline sdk/core/azure-core-management/src/main/java/com/azure/core/management/implementation/polling/PollOperation.java:76]: Here is the bug. Previously, after `doSinglePoll`, the previous `pollResponse` is returned. Should use the latest after `doSinglePoll`.

**Pull request merged**